### PR TITLE
[DRIVER][SYCL] Update the SYCL triple parsing for AMD

### PR DESCRIPTION
**Pull request by jle-quel**
## Description

This PR improves the SYCL target triple parsing for AMD GPUs by enforcing that the triple must be fully specified and explicit.

### CODE REVIEW

File: clang/test/Driver/sycl-oneapi-gpu-amdgpu.cpp [hunk 1]
```diff
@@ -136,6 +136,27 @@
...
+// AMD-TRIPLE: clang{{.*}} "-triple" "amdgcn-amd-amdhsa"
+
+// Check if SYCL triples with 'Environment' component are rejected for AMD GPUs.
+// RUN: not %clangxx -c -fsycl -fsycl-targets=amdgcn-amd-amdhsa-sycl -### %s 2>&1 | \
+  // RUN: FileCheck %s --check-prefix=BAD_TARGET_TRIPLE_ENV
+// RUN: not %clang_cl -c -fsycl -fsycl-targets=amdgcn-amd-amdhsa-sycl -### %s 2>&1 | \
+// RUN:   FileCheck %s --check-prefix=BAD_TARGET_TRIPLE_ENV
+// BAD_TARGET_TRIPLE_ENV: error: SYCL target is invalid: 'amdgcn-amd-amdhsa-sycl'
+
+// Check for invalid SYCL triple for NVidia GPUs.
```
Comment (srividya-sundaram) [inline clang/test/Driver/sycl-oneapi-gpu-amdgpu.cpp:153]: Looks like a copy-paste issue. This should be - AMD GPUs.

File: clang/lib/Driver/Driver.cpp [hunk 2]
```diff
@@ -816,8 +816,10 @@ static bool isValidSYCLTriple(llvm::Triple T) {
       T.getOS() == llvm::Triple::CUDA && !T.hasEnvironment())
     return true;
 
-  // AMDGCN is valid for SYCL
-  if (T.isAMDGCN())
+  // 'amdgcn-amd-amdhsa' is the valid SYCL triple for AMD GPUs.
```
Comment (mdtoguchi) [inline clang/lib/Driver/Driver.cpp:819]: Can https://github.com/intel/llvm/blob/0e3e6d234a94d7d54bba21ae7a7e43f500812b1d/sycl/doc/UsersManual.md?plain=1#L42 be updated to reflect the expected triple to be used here?

Comment (jle-quel) [reply clang/lib/Driver/Driver.cpp:819]: Updated :+1:

**Review by mdtoguchi — APPROVED** [review]

**Review by keryell — APPROVED** [review]
Thanks.
Could you rewrite sycl → SYCL in the PR title since it will be used as the commit message?

**Comment by sommerlukas** [comment]
@intel/dpcpp-doc-reviewers This PR still needs review from you. As far as I can see, it's a single line addition to `UserManual.md` that needs you approval.

**Review by bader — APPROVED** [review]

**Review by srividya-sundaram — APPROVED** [review]

**Pull request merged**